### ISSUE TITLE
 binded arguments in the "path" between function's decorator and parameters' decorators are different

### ISSUE THREAD

**Issue by yadav-saurabh** [samchon/nestia#896]
Seems like Param's definition as an object is not supported, In Nestjs `@Param() params: { id: string; countryId: string }` is valid.
`@TypedParam()` doesn't support it yet. 

```bash
@Get('/:countryId/states/:id')
  findOneState(@Param() params: { id: string; countryId: string }) {
    return this.countriesService.findOneState(+params.countryId, params.id);
  }
```

```text
  - binded arguments in the "path" between function's decorator and parameters' decorators are different (function: [countryId, id], parameters: []).
```

**Comment by samchon** [comment]
@micalevisk Is this intended spec of NestJS? Or just unexpected behavior but just supported by express?

**Comment by micalevisk** [comment]
I don't see anything wrong with that code snippet

**Comment by samchon** [comment]
Got it, it is supported spec from NestJS, but I'll make it to be a compilation error.

**Comment by samchon** [comment]
@yadav-saurabh Is it okay to support object typed `@Param()`, but SDK library decomposes it to individual parameters?

**Comment by yadav-saurabh** [comment]
> @yadav-saurabh Is it okay to support object typed `@Param()`, but SDK library decomposes it to individual parameters?

yes, my initial thinking was the same. They are going be individual parameters from the frontend POV, just nestjs (maybe express) makes an object so that's easier to access if there are multiple parameters.

**Comment by micalevisk** [comment]
> In the object type case, how `@nestjs/swagger` composes the parameters (`OpenApiOperation.parameters`) ?

type defs like that has not effect for `@nestjs/swagger` (the CLI plugin)

**Comment by samchon** [comment]
> > In the object type case, how `@nestjs/swagger` composes the parameters (`OpenApiOperation.parameters`) ?
> 
> type defs like that has not effect for `@nestjs/swagger` (the CLI plugin)

Then should I define `@ApiParam()` to the object typed path parameter, and write the object path parameter type with `@ApiProperty()`?

**Comment by micalevisk** [comment]
it will work if you don't use types but concrete classes, due to how tsc works

or if you use `@ApiParam()`, of course

not sure about the behavior when using both approaches

**Comment by samchon** [comment]
@micalevisk Thaks for help.

@yadav-saurabh To accomplish this mission, I've to disassemble the object typed `@Param()` in the compilation level, and it needs lots of change in both `@nestia/core` and `@nestia/sdk` projects. Furthermore, as `@Param()` is not of my development, but of pure NestJS feature, it may needs lots of time.

By the way, this mission's benefit for users is a little bit lower, so that it can be delayed for a while. 

Until that, recommend to avoid using the object typed `@Param()`.

**Comment by yadav-saurabh** [comment]
> @micalevisk Thaks for help.
> 
> @yadav-saurabh To accomplish this mission, I've to disassemble the object typed `@Param()` in the compilation level, and it needs lots of change in both `@nestia/core` and `@nestia/sdk` projects. Furthermore, as `@Param()` is not of my development, but of pure NestJS feature, it may needs lots of time.
> 
> By the way, this mission's benefit for users is a little bit lower, so that would be delayed due to lower priority.
> 
> Until that, recommend to avoid using the object typed `@Param()`.

Yes, I can understand.

Thanks for the quick response

**Comment by samchon** [comment]
Will start this ticket after #211

**Comment by samchon** [comment]
As I'm concentrating AI development using Nestia, and vibe coding agent generating NestJs/Nestia app, I can't do this mission.
